packages/starknet-snap/src/index.tsx

```diff
@@ -302,42 +297,38 @@ export const onRpcRequest: OnRpcRequestHandler = async ({ request }) => {
 };
 
 export const onInstall: OnInstallHandler = async () => {
-  await snap.request({
-    method: 'snap_dialog',
-    params: {
-      type: 'alert',
-      content: (
-        <Box>
-          <Text>Your MetaMask wallet is now compatible with Starknet!</Text>
-          <Text>
-            To manage your Starknet account and send and receive funds, visit
-            the <Link href={getDappUrl()}>companion dapp for Starknet</Link>.
-          </Text>
-        </Box>
-      ),
-    },
-  });
+  const initSnapStateManager = new InitSnapStateManager();
+  await initSnapStateManager.ensureJsxSupport(
+    <Box>
+      <Text>Your MetaMask wallet is now compatible with Starknet!</Text>
+      <Text>
+        To manage your Starknet account and send and receive funds, visit the{' '}
+        <Link href={getDappUrl()}>companion dapp for Starknet</Link>.
+      </Text>
+    </Box>,
+  );
 };
 
 export const onUpdate: OnUpdateHandler = async () => {
-  await snap.request({
-    method: 'snap_dialog',
-    params: {
-      type: 'alert',
-      content: (
-        <Box>
-          <Text>Your Starknet Snap is now up-to-date !</Text>
-          <Text>
-            As usual, to manage your Starknet account and send and receive
-            funds, visit the{' '}
-            <Link href={getDappUrl()}>companion dapp for Starknet</Link>.
-          </Text>
-        </Box>
-      ),
-    },
-  });
+  const initSnapStateManager = new InitSnapStateManager();
+  await initSnapStateManager.ensureJsxSupport(
+    <Box>
+      <Text>Your Starknet Snap is now up-to-date !</Text>
+      <Text>
+        As usual, to manage your Starknet account and send and receive funds,
+        visit the <Link href={getDappUrl()}>companion dapp for Starknet</Link>.
+      </Text>
+    </Box>,
+  );
 };
 
 export const onHomePage: OnHomePageHandler = async () => {
+  const initSnapStateManager = new InitSnapStateManager();
+  const requireMMUpgrade = await initSnapStateManager.requireMetaMaskUpgrade();
```

Comment:
it will cause racing condition, 
homepage and oninstall and onupdate are running at same time

and i think we can do the same with onUpdate and onInstall